torchtune/utils/metric_logging.py

```diff
@@ -317,3 +317,127 @@ def close(self) -> None:
         if self._writer:
             self._writer.close()
             self._writer = None
+
+
+class CometLogger(MetricLoggerInterface):
+    """Logger for use w/ Comet (https://www.comet.com/site/).
+    For more information about arguments expected by Comet, see
+    https://www.comet.com/docs/v2/guides/experiment-management/configure-sdk/#for-the-experiment.
+
+    Args:
+        api_key (Optional[str]): Comet API key. It's recommended to configure the API Key from the environment.
+        workspace (Optional[str]): Comet workspace name. If not provided, uses the default workspace.
+        project (Optional[str]): Comet project name. Defaults to Uncategorized.
+        experiment_name (Optional[str]): The name for comet experiment to be used for logging.
+        experiment_key (Optional[str]): The key for comet experiment to be used for logging. Must be an alphanumeric
+            string whose length is between 32 and 50 characters.
+        mode (Optional[str]): Control how the Comet experiment is started. "get": Continue logging to an existing
+            experiment identified by the `experiment_key` value. "create": Always creates of a new experiment, useful
+            for HPO sweeps. "get_or_create" (default): Starts a fresh experiment if required, or persists logging to
+            an existing one.
+        online (Optional[bool]): If True, the data will be logged to Comet server, otherwise it will be stored locally
+            in offline experiment. Default is `True`.
+        experiment_name (Optional[str]): Name of the experiment. If not provided, Comet will auto-generate a name.
+        tags (Optional[List[str]]): Tags to associate with the experiment.
+        log_code (bool): Whether to log the source code. Defaults to True.
+        **kwargs: additional arguments to pass to Comet.start. See
```

Comment:
might need to type this (annoyingly)
```suggestion
        **kwargs: additional arguments to pass to Comet.start. See
```